semgrep-core/src/engine/Match_rules.ml

```diff
@@ -796,48 +796,90 @@ and filter_ranges env xs cond =
          | R.CondGeneric e ->
              let env = Eval_generic.bindings_to_env bindings in
              Eval_generic.eval_bool env e
-         | R.CondPattern (mvar, formula) -> (
+         | R.CondPattern (mvar, opt_lang, formula) -> (
              (* If anything goes wrong the default is to filter out! *)
+             let lazy_ast_of_string lang str =
+               lazy
+                 (let ext =
+                    match Lang.ext_of_lang lang with
+                    | x :: _ -> x
+                    | [] -> assert false
+                  in
+                  Common2.with_tmp_file ~str ~ext (fun file ->
+                      let { Parse_target.ast; errors; _ } =
+                        Parse_target
+                        .parse_and_resolve_name_use_pfff_or_treesitter lang file
+                      in
+                      (ast, errors)))
+             in
+             let go_eval_formula env xlang formula lazy_ast_and_errors
```

Comment:
Same here, because it's difficult to see if it's use enclosed variables (which then are really like using globals).